File: app/src/main/java/com/temoa/gankio/network/GankApi.java
package com.temoa.gankio.network;

import com.temoa.gankio.bean.NewGankData;

import retrofit2.http.GET;
import retrofit2.http.Path;
import rx.Observable;

/**
 * Created by Temoa
 * on 2016/8/2 13:45
 */
public interface GankApi {

    @GET("{type}/{count}/{pageIndex}")
    Observable<NewGankData> getNewData(
            @Path("type") String type,
            @Path("count") int count,
            @Path("pageIndex") int pageIndex
    );

    @GET("福利/3/1")
    Observable<NewGankData> getBeauty();
}

File: app/src/main/java/com/temoa/gankio/network/GankApi.kt
package com.temoa.gankio.network

import com.temoa.gankio.bean.NewGankData
import io.reactivex.Observable
import retrofit2.http.GET
import retrofit2.http.Path

/**
 * Created by Temoa
 * on 2016/8/2 13:45
 */
interface GankApi {

  @GET("{type}/{count}/{pageIndex}")
  fun getNewData(
      @Path("type") type: String?,
      @Path("count") count: Int,
      @Path("pageIndex") pageIndex: Int
  ): Observable<NewGankData>

  @get:GET("福利/3/1")
  val beauty: Observable<NewGankData>
}